plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -42,24 +45,31 @@
     public final static String INDEX = ".pipelines";
     public final static String TYPE = "pipeline";
 
-    private Client client;
-    private final Injector injector;
+    private final Updater updater;
+    private final ClusterService clusterService;
+    private final TimeValue pipelineUpdateInterval;
+    private final PipelineConfigDocReader configDocReader;
+    private final Map<String, Processor.Builder> processorRegistry;
 
-    private volatile Updater updater;
-    private volatile CopyOnWriteHashMap<String, Pipeline> pipelines = new CopyOnWriteHashMap<>();
+    private volatile CopyOnWriteHashMap<String, PipelineReference> pipelines = new CopyOnWriteHashMap<>();
```

Comment:
Personally I don't see the added value here as the map is never written to... not sure how more efficient it is on retrieval side compared to normal hash map... also on memory side as we don't expect to store here that many pipelines? maybe @jpountz can comment here. In any case, the variable should be of type `Map<String, PipelineReference>` the implementation is another discussion.